SCons/Environment.py

```diff
@@ -123,7 +123,7 @@ def apply_tools(env, tools, toolpath):
 
 def copy_non_reserved_keywords(dict):
     result = semi_deepcopy(dict)
-    for k in list(result.keys()):
+    for k in result.copy().keys():
```

Comment:
why .copy().keys()?

Reply:
An abundance of caution: it's bad form to modify an iterable while iterating over it. I don't have proof it causes a problem. Py3 porting thing: in Py2, the original line would be looping over the list created by getting the keys from result - safe. In Py3 you get a dictionary view - that is, you're looping over the original dict.

Reply:
in this case it's clearly not modifying the list..

Reply:
since the original forced to a list, no. could leave it alone.

Reply:
ahh.sorry didn't see the del.. below.